packages/react/src/components/cookieConsent/cookieModal/CookieModal.test.tsx

```diff
@@ -0,0 +1,331 @@
+
+const mockedCookieControls = mockDocumentCookie();
+
+let content: Content;
+
+const renderCookieConsent = (
+  { requiredConsents = [], optionalConsents = [], consents = {}, contentSourceOverrides }: TestConsentData,
+  withRealTimers = false,
+): RenderResult => {
+  // inject unknown consents to verify those are
+  // stored and handled, but not required or optional
+  const consentCookieWithInjectedUnknowns = {
+    ...consents,
+    ...unknownConsents,
+  };
+  const contentSource = getContentSource(requiredConsents, optionalConsents, contentSourceOverrides);
+  content = createContent(contentSource);
+  jest.useFakeTimers();
+  mockedCookieControls.init({ [COOKIE_NAME]: JSON.stringify(consentCookieWithInjectedUnknowns) });
+  const result = render(
+    <div>
+      <CookieModal contentSource={contentSource} />
+      <button id="focus-target" type="button">
+        Focus me
+      </button>
+    </div>,
+  );
+  act(() => {
+    jest.runAllTimers();
+  });
+
+  // For example, axe uses timers so sometimes the test must use real ones
+  if (withRealTimers) {
+    jest.useRealTimers();
+  }
+
+  return result;
+};
+
+describe('<ModalContent /> spec', () => {
+  afterEach(() => {
+    mockedCookieControls.clear();
+  });
+
+  afterAll(() => {
+    mockedCookieControls.restore();
+  });
+
+  it('renders the component', () => {
+    const { baseElement } = renderCookieConsent(defaultConsentData);
+    expect(baseElement).toMatchSnapshot();
+  });
+
+  it('should not have basic accessibility issues', async () => {
+    const result = renderCookieConsent(defaultConsentData, true);
+    await openAccordion(result, dataTestIds.settingsToggler);
+    await openAllAccordions(result, content, dataTestIds);
+    const results = await axe(result.container);
+    expect(results).toHaveNoViolations();
+  }, 15000);
+});
+
+describe('<Modal /> ', () => {
+  afterEach(() => {
+    mockedCookieControls.clear();
+  });
+
+  afterAll(() => {
+    mockedCookieControls.restore();
+  });
+
+  const getSetCookieArguments = (index = -1) => extractSetCookieArguments(mockedCookieControls, index);
+
+  const initDetailsView = async (data: TestConsentData): Promise<RenderResult> => {
+    const result = renderCookieConsent(data);
+    await openAccordion(result, dataTestIds.settingsToggler);
+    return result;
+  };
+
+  describe('Cookie consent ', () => {
+    it('and child components are rendered when consents have not been handled', () => {
+      const result = renderCookieConsent(defaultConsentData);
+      verifyElementExistsByTestId(result, dataTestIds.container);
+      verifyElementExistsByTestId(result, dataTestIds.languageSwitcher);
+      verifyElementExistsByTestId(result, dataTestIds.approveButton);
+      verifyElementExistsByTestId(result, dataTestIds.approveRequiredButton);
+    });
+
+    it('is rendered if a required consent has not been approved. It could have been optional before', () => {
```

Comment:
If there's only required cookies, then I thought it was not necessary to show the cookie consent modal then?

Reply:
In this case the test data has optional cookies. Should clarify that, or change test case where only required cookies and one is false.

Reply:
CookieModal.test.tsx is removed in another PR. Replaced by PortalModal